### PR TITLE
chore: normalize buttons

### PR DESCRIPTION
**Pull request by luchaos**
One of the first components PRs extracted from the prototype/preview branch.

Normalize button usage:
- No `<input type="submit">`
- No `<button type="submit">` (`submit` is implicit)
- No implicit styling of `<button>` (relevant for upcoming PRs)
- Buttons with JS handlers use `type="button"`

That's the third or fourth time I'm normalizing those buttons now 🤔 
I hope this time it sticks 😅

### CODE REVIEW

File: resources/views/components/modal-trigger.blade.php [hunk 1]
```diff
@@ -39,13 +39,13 @@ class="bg-box-bg relative w-full h-screen max-h-screen overflow-y-auto sm:h-auto
             >
                 <div class="flex items-center justify-between pb-2">
                     <p class="text-lg font-semibold mb-4">{{ $modalTitleLabel }}</p>
-                    <button
```
Comment (luchaos) [inline resources/views/components/modal-trigger.blade.php:42]: It looks like VS Code ignores the editorconfig for trimming whitespace?
https://github.com/RetroAchievements/RAWeb/blob/master/.editorconfig#L9
`trim_trailing_whitespace = true`

File: resources/css/button.css [hunk 2]
```diff
@@ -1,4 +1,4 @@
-.btn, button, [type="button"], [type="reset"], [type="submit"] {
```
Comment (luchaos) [inline resources/css/button.css:1]: While popular in CSS frameworks like bootstrap and Daisy UI (where this is from) it's too greedy for our use

File: app/Helpers/render/layout.php [hunk 3]
```diff
@@ -261,7 +261,7 @@ function RenderToolbar(): void
     echo "<form class='flex searchbox-top' action='/searchresults.php'>";
     // echo "Search:&nbsp;";
     echo "<input name='s' type='text' class='flex-1 searchboxinput' value='$searchQuery' placeholder='Search the site...'>";
-    echo "<input type='submit' value='🔎︎' title='Search the site'>";
+    echo "<button class='btn' title='Search the site'>🔎</button>";
```
Comment (wescopeland) [inline app/Helpers/render/layout.php:264]: Adding a border to this button probably isn't desirable.

Prod:
Screenshot 2023-08-17 at 5 40 53 PM


PR:
Screenshot 2023-08-17 at 5 40 10 PM

File: public/gameInfo.php
Comment (wescopeland) [inline public/gameInfo.php:1]: The 3rd button on the page was missed:

Screenshot 2023-08-17 at 5 44 41 PM

File: resources/views/components/modal-trigger.blade.php (hunk 1 above)
Comment (wescopeland) [reply resources/views/components/modal-trigger.blade.php:42]: Confirmed, VSCode does indeed ignore _.editorconfig_ unless a third-party extension is installed.

File: resources/css/button.css (hunk 2 above)
Comment (wescopeland) [reply resources/css/button.css:1]: We have a few of these global style overrides lurking, hopefully we can systematically remove them over time.

File: public/gameInfo.php
Comment (luchaos) [reply public/gameInfo.php:1]: Good catch! Fixed in https://github.com/RetroAchievements/RAWeb/pull/1774/commits/8bbcf4791a54bd8d54be2004ed3eb0ea4a955432

File: app/Helpers/render/layout.php (hunk 3 above)
Comment (luchaos) [reply app/Helpers/render/layout.php:264]: Good catch! Fixed in https://github.com/RetroAchievements/RAWeb/pull/1774/commits/8bbcf4791a54bd8d54be2004ed3eb0ea4a955432

**Review by wescopeland — APPROVED** [review]

**Pull request merged**